File: a916f5fb-11b1-498d-8a70-78c17b4a2e2d/src/main/java/com/tobacco/controller/AuthController.java
package com.tobacco.controller;

import com.tobacco.common.result.Result;
import com.tobacco.dto.request.LoginRequest;
import com.tobacco.dto.response.LoginResponse;
import com.tobacco.service.AuthService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

@Tag(name = "认证授权", description = "用户登录、令牌刷新、登出接口")
@RestController
@RequestMapping("/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @Operation(summary = "用户登录", description = "使用用户名和密码登录系统，获取JWT访问令牌")
    @PostMapping("/login")
    public Result<LoginResponse> login(@Valid @RequestBody LoginRequest request) {
        return Result.success(authService.login(request));
    }

    @Operation(summary = "刷新令牌", description = "使用刷新令牌获取新的访问令牌")
    @PostMapping("/refresh")
    public Result<LoginResponse> refreshToken(@RequestBody String refreshToken) {
        return Result.success(authService.refreshToken(refreshToken));
    }

    @Operation(summary = "用户登出", description = "退出登录，使当前令牌失效")
    @PostMapping("/logout")
    public Result<Void> logout() {
        authService.logout();
        return Result.success();
    }
}

File: 9096d8e2-d12b-468a-91ae-53c97e7bf587/backend/src/main/kotlin/com/ems/dispatch/controller/AuthController.kt
package com.ems.dispatch.controller

import com.ems.dispatch.dto.ApiResponse
import com.ems.dispatch.repository.UserRepository
import com.ems.dispatch.security.JwtService
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.responses.ApiResponse
import io.swagger.v3.oas.annotations.responses.ApiResponses
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import org.springframework.http.ResponseEntity
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
@Tag(name = "认证管理", description = "用户登录、Token刷新相关API")
class AuthController(
    private val authenticationManager: AuthenticationManager,
    private val userDetailsService: UserDetailsService,
    private val jwtService: JwtService,
    private val userRepository: UserRepository
) {
    data class LoginRequest(
        val username: String,
        val password: String
    )

    data class LoginResponse(
        val accessToken: String,
        val tokenType: String = "Bearer",
        val expiresIn: Int = 86400,
        val username: String,
        val realName: String,
        val role: String
    )

    @PostMapping("/login")
    @Operation(
        summary = "用户登录",
        description = "使用用户名和密码登录获取JWT Token"
    )
    @ApiResponses(
        ApiResponse(responseCode = "200", description = "登录成功"),
        ApiResponse(responseCode = "401", description = "用户名或密码错误")
    )
    fun login(@Valid @RequestBody request: LoginRequest): ResponseEntity<ApiResponse<LoginResponse>> {
        val authentication = authenticationManager.authenticate(
            UsernamePasswordAuthenticationToken(request.username, request.password)
        )

        val userDetails = userDetailsService.loadUserByUsername(request.username)
        val token = jwtService.generateToken(emptyMap(), userDetails)
        val user = userRepository.findByUsername(request.username)!!

        val response = LoginResponse(
            accessToken = token,
            username = user.username,
            realName = user.realName,
            role = user.role
        )

        return ResponseEntity.ok(ApiResponse.success(response, "登录成功"))
    }
}
